src/azul/service/elasticsearch_service.py

```diff
@@ -369,39 +369,40 @@ def _apply_paging(self,
         """
         # Extract the fields for readability (and slight manipulation)
 
-        _sort = pagination['sort'] + '.keyword'
-        _order = pagination['order']
+        sort_field = pagination['sort'] + '.keyword'
+        sort_order = pagination['order']
 
         field_type = self.field_type(catalog, tuple(pagination['sort'].split('.')))
-        _mode = field_type.es_sort_mode
+        sort_mode = field_type.es_sort_mode
 
-        def sort_values(sort_field, sort_order, sort_mode):
-            assert sort_order in ('asc', 'desc'), sort_order
+        def sort(order):
+            assert order in ('asc', 'desc'), order
             return (
                 {
                     sort_field: {
-                        'order': sort_order,
+                        'order': order,
                         'mode': sort_mode,
-                        'missing': '_last' if sort_order == 'asc' else '_first'
+                        'missing': '_last' if order == 'asc' else '_first',
+                        'unmapped_type': field_type.es_type
```

Comment:
Is there documented evidence that ES supports `unmapped_type: null`? Your tests explicitly skip this case.

Reply:
Omitting `unmapped_type` if `es_type == None`.
Note that `assay_type` and `organismAgeRange` will not work correctly and will be addressed in #2621